tasks.py

```diff
@@ -1643,3 +1771,182 @@ def get_command_version(get_version_command: str, version_prefix: str) -> Option
 def is_ipv4(addr):
     ip = ipaddress.ip_network(addr)
     return ip.version == 4
+
+
+def is_ipv6(addr):
+    ip = ipaddress.ip_network(addr)
+    return ip.version == 6
+
+
+def detect_bgp_type(namespace="metallb-system"):
+    """Detect BGP type from controller deployment."""
+    bgp_type_result = run(
+        f"{kubectl_path} get deployment controller -n {namespace} "
+        f"-o jsonpath='{{.spec.template.spec.containers[0].env[?(@.name==\"METALLB_BGP_TYPE\")].value}}'",
+        hide=True,
+        warn=True,
+    )
+
+    if bgp_type_result.ok:
+        value = bgp_type_result.stdout.strip()
+        # 'native' if not set (follows the Go code behaviour)
+        return value if value else "native"
+    else:
+        raise Exception(
+            f"Cannot detect BGP type from controller deployment in {namespace} namespace: command failed"
+        )
+
+
+def detect_prometheus():
+    """Detect if Prometheus is available by checking monitoring namespace."""
+    prom_result = run(f"{kubectl_path} get namespace monitoring", hide=True, warn=True)
+    return prom_result.ok
+
+
+def detect_ip_family(namespace="metallb-system"):
+    """Detect IP family from ipaddresspool resource (dual, ipv4, ipv6)."""
```

Comment:
there is no guarantee (or reason) that an ipaddresspool exists in a clean cluster, I think we should opt to checking the node/pods internal ips (something like https://kubernetes.io/docs/tasks/network/validate-dual-stack/#validate-node-addressing)